functions/src/index.ts

```diff
@@ -0,0 +1,231 @@
+import * as axios from 'axios';
+import * as base64 from 'js-base64'
+import * as functions from 'firebase-functions';
+import * as  admin from 'firebase-admin';
+// Will need to set up admin credentials for first time use
+admin.initializeApp({
+    credential: admin.credential.applicationDefault(),
+    databaseURL: "https://tuneiq.firebaseio.com"
+});
+
+// Types -- TODO: find a better place to put these
+export type RoundDetails = {
+    url: string
+    tracks: GameTrack[]
+}
+
+export type GameTrack = {
+    id: string,
+    title: string,
+    artists: string[],
+    isAnswer: boolean,
+}
+
+export type GameDetails = {
+    name: string,
+    password: string,
+    hostId: string,
+    intermissionDuration: number,
+    roundDuration: number,
+    startTime: Date | null,
+    genre: string,
+    rounds: RoundDetails[],
+    leaderBoard: {
+        [id: string]: {
+            name: string,
+            score: number
+        }
+    }
+}
+
+export type SpotifyArtist = {
+    id: string,
+    name: string,
+    //The Spotify URI for the artist
+    uri: string
+}
+
+export type SpotifyTrack = {
+    id: string,
+    artists: SpotifyArtist[],
+    duration_ms: number,
+    explict: boolean,
+    name: string,
+    preview_url: string,
+    uri: string,
+}
+
+// Functions
+exports.createGame = functions.https.onRequest((request, response) => {
+    const gameGenre: string = String(request.query.genre)
+    const hostId: string = String(request.query.hostId)
+    const hostName: string = String(request.query.hostName)
+    const gameName: string = String(request.query.name)
+    const password: string = String(request.query.password)
+    const gameRounds: number = Number(functions.config().gameconfig.number_of_rounds)
+    const optionsPerRound: number = Number(functions.config().gameconfig.number_of_options)
+
+    getRoundDetails(gameGenre, gameRounds, optionsPerRound).then((roundDetails: RoundDetails[]) => {
+        const gameDetails: GameDetails = {
+            hostId: hostId,
+            name: gameName,
+            password: password,
+            genre: gameGenre,
+            rounds: roundDetails,
+            intermissionDuration: NaN,
+            roundDuration: NaN,
+            startTime: null,
+            leaderBoard: {
```

Comment:
leaderboard should be an array of object no? or is this the right syntax for that?

Reply:
it could be, I thought an object would be better for look up. Not sure if its the correct way